contrib/epee/src/wipeable_string.cpp

```diff
@@ -109,9 +115,112 @@ void wipeable_string::push_back(char c)
   buffer.back() = c;
 }
 
-void wipeable_string::pop_back()
+void wipeable_string::operator+=(char c)
+{
+  push_back(c);
+}
+
+void wipeable_string::operator+=(const std::string &s)
+{
+  CHECK_AND_ASSERT_THROW_MES(size() < std::numeric_limits<size_t>::max() - s.size(), "Appended data too large");
```

Comment:
IIRC we found out std::string can't be wiped, because the standard doesn't get us a pointer to a writeable buffer. Hence wipeable_string using a buffer rather than a string.
If you want to wipe data, then you do not use a string in the first place.
These extra dupes are needed due to the seed code.

Reply:
Unfortunately `s.data()` is only available in C++17+, but `&s[0]` provides access to the buffer. The implementation could perform a copy in that call, but is unlikely to do so now that copy-on-write is not allowed by the standard since C++11.

The more critical question that you avoided - why bother with `wipeable_string` if there is an existing heap buffer with the same contents that is never wiped? The use of `wipeable_string` is for "appearances only" in that situation. It would seem that either a `std::string&&` + wipe should be attempted OR just punt entirely on the extra complications of using this class.

Reply:
See the comment about std::map or unordered_map for the words list. The words list is not secret, but the strings being looked up are, so the map needs to have wipeable strings in.